storage/pebble/registers_test.go

```diff
@@ -12,117 +13,193 @@ import (
+// TestRegisters_Store_RoundTrip tests the round trip of a payload storage.
+func TestRegisters_Store_RoundTrip(t *testing.T) {
+	t.Parallel()
+	minHeight := uint64(2)
+	RunWithRegistersStorageAtInitialHeights(t, minHeight, minHeight, func(r *Registers) {
+		key1 := flow.RegisterID{Owner: "owner", Key: "key1"}
+		expectedValue1 := []byte("value1")
+		entries := flow.RegisterEntries{
+			{Key: key1, Value: expectedValue1},
+		}
+		testHeight := minHeight + 1
+		// happy path
+		err := r.Store(testHeight, entries)
+		require.NoError(t, err)
+
+		// lookup with exact height returns the correct value
+		value1, err := r.Get(testHeight, key1)
+		require.NoError(t, err)
+		require.Equal(t, expectedValue1, value1)
+
+		value11, err := r.Get(testHeight, key1)
+		require.NoError(t, err)
+		require.Equal(t, expectedValue1, value11)
+	})
+}
 
-	value1, err := s.Get(height1, key1)
-	require.NoError(t, err)
-	require.Equal(t, expectedValue1, value1)
+// TestRegisters_Store_Versioning tests the scan functionality for the most recent value
+func TestRegisters_Store_Versioning(t *testing.T) {
+	t.Parallel()
+	RunWithRegistersStorageAtHeight1(t, func(r *Registers) {
+		// Save key11 is a prefix of the key1, and we save it first.
+		// It should be invisible for our prefix scan.
+		key11 := flow.RegisterID{Owner: "owner", Key: "key11"}
+		expectedValue11 := []byte("value11")
+
+		key1 := flow.RegisterID{Owner: "owner", Key: "key1"}
+		expectedValue1 := []byte("value1")
+		entries1 := flow.RegisterEntries{
+			{Key: key1, Value: expectedValue1},
+			{Key: key11, Value: expectedValue11},
+		}
 
-	value1, err = s.Get(height3-1, key1)
-	require.NoError(t, err)
-	require.Equal(t, expectedValue1, value1)
+		height2 := uint64(2)
 
-	// test new version
-	value1, err = s.Get(height3, key1)
-	require.NoError(t, err)
-	require.Equal(t, expectedValue1ge3, value1)
+		// check increment in height after Store()
+		err := r.Store(height2, entries1)
+		require.NoError(t, err)
 
-	value1, err = s.Get(height3+1, key1)
-	require.NoError(t, err)
-	require.Equal(t, expectedValue1ge3, value1)
+		// Add new version of key1.
+		height3 := uint64(3)
+		expectedValue1ge3 := []byte("value1ge3")
+		entries3 := flow.RegisterEntries{
+			{Key: key1, Value: expectedValue1ge3},
+		}
 
-	err = db.Close()
-	require.NoError(t, err)
+		// check increment in height after Store()
+		err = r.Store(height3, entries3)
+		require.NoError(t, err)
+		updatedHeight := r.LatestHeight()
+		require.Equal(t, updatedHeight, height3)
+
+		// test old version at previous height
+		value1, err := r.Get(height2, key1)
+		require.NoError(t, err)
+		require.Equal(t, expectedValue1, value1)
+
+		// test new version at new height
+		value1, err = r.Get(height3, key1)
+		require.NoError(t, err)
+		require.Equal(t, expectedValue1ge3, value1)
+
+		// test unchanged key at incremented height
+		value11, err := r.Get(height3, key11)
+		require.NoError(t, err)
+		require.Equal(t, expectedValue11, value11)
+	})
```

Comment:
better to add a case that a newly added key does not exist in the previous height.